tests/test_sendinblue_backend.py

```diff
@@ -0,0 +1,444 @@
+            self.message.send_at = datetime(2016, 3, 4, 5, 6, 7, tzinfo=utc_minus_8)
+
+            with self.assertRaises(AnymailUnsupportedFeature):
+                self.message.send()
+
+    def test_tag(self):
+        self.message.tags = ["receipt"]
+        self.message.send()
+        data = self.get_api_call_json()
+        self.assertCountEqual(data['headers']["X-Mailin-tag"], "receipt")
+
+    def test_multiple_tags(self):
+        self.message.tags = ["receipt", "repeat-user"]
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_tracking(self):
+        # Test one way...
+        self.message.track_clicks = False
+        self.message.track_opens = True
+
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+        # ...and the opposite way
+        self.message.track_clicks = True
+        self.message.track_opens = False
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_template_id(self):
+        # SendinBlue use incremental ID to identify templates
+        self.message.template_id = "12"
+        self.message.send()
+
+        # We can't verify if the Url has changed
+
+    def test_template_id_with_empty_body(self):
+        message = mail.EmailMessage(from_email='from@example.com', to=['to@example.com'])
+        message.template_id = "9"
+        message.send()
+        data = self.get_api_call_json()
+        self.assertNotIn('htmlcontent', data)
+        self.assertEquals(data['textContent'], '')  # neither text nor html body
+        self.assertNotIn('subject', data)
+
+    def test_merge_data(self):
+        self.message.from_email = 'from@example.com'
+        self.message.to = ['alice@example.com', 'Bob <bob@example.com>', 'celia@example.com']
+        self.message.cc = ['cc@example.com']
+        self.message.body = "Hi :name. Welcome to :group at :site."
+        self.message.merge_data = {
+            'alice@example.com': {':name': "Alice", ':group': "Developers"},
+            'bob@example.com': {':name': "Bob"},  # and leave :group undefined
+        }
+        self.message.merge_global_data = {
+            ':group': "Users",
+            ':site': "ExampleCo",
+        }
+
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_default_omits_options(self):
+        """Make sure by default we don't send any ESP-specific options.
+
+        Options not specified by the caller should be omitted entirely from
+        the API call (*not* sent as False or empty). This ensures
+        that your ESP account settings apply by default.
+        """
+        self.message.send()
+        data = self.get_api_call_json()
+        self.assertNotIn('attachment', data)
+        self.assertNotIn('tag', data)
+        self.assertNotIn('headers', data)
+        self.assertNotIn('send_at', data)
```

Comment:
This should assertNotIn for all the optional ESP API params. So, it's not necessary to check `'send_at'` (not a SiB param), but it'd be good to check `'replyTo'` and maybe `'attributes'`.

Reply:
Done